Source/MediaStorage.swift

```diff
@@ -364,4 +371,94 @@ public class MediaStorage {
+        } else {
+            userMetaKey = replaceUserMeta(fieldName)
+            if userMetaKey == "" {
+                completionHandler(Dictionary<String, String>(), MediaStorageError(statusCode: nil, message: "invalid parameter: \(fieldName)"))
+                return
+            } else {
+                // GET /media/{id}/meta/user/{key}
+                url = "\(mstorageEndpoint)/\(mediaId)\(getUserMetaPath)/\(userMetaKey)"
+                isUserKey = true
+            }
+        }
+
+        MediaStorageRequest.get(
+            url: url,
+            queryParams: [String: String](),
+            header: [
+                "Authorization" : "Bearer \(accessToken!)"
+            ]
+        ){(data, resp, err) in
+            if err != nil {
+                completionHandler(Dictionary<String, String>(), MediaStorageError(statusCode: nil, message: "request failed: \(err!.code): \(err!.domain)"))
+                return
+            }
+
+            let httpresp = resp as! NSHTTPURLResponse
+            let statusCode = httpresp.statusCode
+            let dataString = NSString(data: data!, encoding: NSUTF8StringEncoding) as! String
+            if !httpresp.isSucceeded() {
+                completionHandler(
+                    Dictionary<String, String>(),
+                    MediaStorageError(statusCode: statusCode, message: "received error: \(dataString)")
+                )
+                return
+            }
+
+            var dataDic = Dictionary<String, String>()
+            if isUserKey {
+                dataDic[userMetaKey] = dataString
+
+            } else  {
+                do {
+                    dataDic = try NSJSONSerialization.JSONObjectWithData(data!, options: NSJSONReadingOptions.AllowFragments) as! [String : String]
+                } catch {
+                    completionHandler(
+                        Dictionary<String, String>(),
+                        MediaStorageError(statusCode: statusCode, message: "invalid response: \(dataString)")
+                    )
+                }
+
+            }
+
+            completionHandler(
+                dataDic,
+                MediaStorageError(statusCode: nil, message: nil)
+            )
+
+        }
+    }
+
+    private func replaceUserMeta (userMeta: String!) -> String{
+        var regex: NSRegularExpression
+        do {
+            let pattern = replaceUserMetaRegex
+            regex = try NSRegularExpression(pattern: pattern, options: NSRegularExpressionOptions.AllowCommentsAndWhitespace)
+            let matches = regex.matchesInString(userMeta, options: NSMatchingOptions.ReportProgress, range:NSMakeRange(0, userMeta.characters.count))
+            if matches.count > 0 {
+                return userMeta.substringFromIndex(userMeta.startIndex.advancedBy(5))
+            } else {
+                return ""
+            }
+        } catch {
+            return ""
+        }
```

Comment:
I think following flow is more simpler. How about you?
```swift
let replace = "$1"
let replaceString = userMeta.stringByReplacingOccurrencesOfString(replaceUserMetaRegex, withString: replace, options: NSStringCompareOptions.RegularExpressionSearch, range: nil)

return replaceString == userMeta ? "" : replaceString
```

I tried the above code in Playground.
<img width="1072" alt="2016-07-12 15 46 35" src="https://cloud.githubusercontent.com/assets/1768970/16757561/d5d02634-4847-11e6-9f60-62953526f707.png">